vllm/model_executor/layers/determinism.py

```diff
@@ -0,0 +1,572 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
Please cite the original source (I guess it's https://github.com/thinking-machines-lab/batch_invariant_ops/blob/main/batch_invariant_ops/batch_invariant_ops.py)

Reply:
In addition, I think it's better practice to copy paste the exact file, and then write another file work as the vLLM wrapper for that file.